rust/src/extensions.rs

```diff
@@ -0,0 +1,191 @@
+//! Core logic for extensions.yaml file.
+
+/*
+ * Copyright (C) 2020 Red Hat, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ */
+
+use anyhow::{bail, Context, Result};
+use openat_ext::OpenatDirExt;
+use serde_derive::Deserialize;
+use std::collections::HashMap;
+use std::io::Read;
+
+use crate::ffi::StringMapping;
+use crate::utils;
+
+const RPMOSTREE_EXTENSIONS_STATE_FILE: &str = ".rpm-ostree-state-chksum";
+
+#[derive(Deserialize, Debug)]
+pub struct Extensions {
+    extensions: HashMap<String, Extension>,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct Extension {
+    packages: Vec<String>,
+    architectures: Option<Vec<String>>,
+    #[serde(rename = "match-base-evr")]
```

Comment:
I generally prefer using `#[serde(rename-all = "kebab-case")]` on the whole struct so it's enforced consistent.